src/session/__tests__/sessionReducer-test.js

```diff
@@ -38,4 +38,39 @@ describe('sessionReducer', () => {
     const newState = sessionReducer(baseState, eg.action.login_success);
     expect(newState).toEqual({ ...baseState, needsInitialFetch: true });
   });
+
+  test('DEAD_QUEUE', () => {
+    const newState = sessionReducer(baseState, eg.action.dead_queue);
+    expect(newState).toEqual({ ...baseState, needsInitialFetch: true });
+  });
+
+  test('LOGOUT', () => {
+    const newState = sessionReducer(baseState, eg.action.logout);
+    expect(newState).toEqual(baseState);
+  });
+
+  test('REALM_INIT', () => {
+    const newState = sessionReducer(baseState, eg.action.realm_init);
+    expect(newState).toEqual({ ...baseState, eventQueueId: 1 });
+  });
+
+  test('DO_NARROW', () => {
+    const newState = sessionReducer(baseState, eg.action.do_narrow);
+    expect(newState).toEqual({ ...baseState, lastNarrow: privateNarrow('a@a.com') });
+  });
+
+  test('APP_ONLINE', () => {
+    const newState = sessionReducer(baseState, eg.action.app_online);
+    expect(newState).toEqual({ ...baseState, isOnline: true });
+  });
+
+  test('APP_STATE', () => {
+    const newState = sessionReducer(baseState, eg.action.app_state);
+    expect(newState).toEqual({ ...baseState, isActive: true });
+  });
```

Comment:
In each of these, notice that the test can't be understood by reading just the test -- it's dependent on the specific details of `eg.action.do_narrow` and so on.

E.g., why should we have `isOnline: true`? That's not a general fact about any `APP_ONLINE` action -- it's only when the action carries a `true` value.

By contrast in the existing `LOGIN_SUCCESS` test, notice that it completely doesn't matter for this test (or the code it's testing) what the specific data in the action is -- only that it's some `LOGIN_SUCCESS` action. That's the case where `exampleData` is super useful: providing some valid value of a given type, for tests where the details of the value don't matter.

For these tests where the specific data does matter, see the `START_EDIT_MESSAGE` test for a good pattern to follow.